src/main/java/net/minecraftforge/registries/DeferredRegister.java

```diff
@@ -108,7 +156,7 @@ public void register(IEventBus bus)
 
     private void addEntries(RegistryEvent.Register<?> event)
     {
-        if (event.getGenericType() == this.type.getRegistrySuperType())
+        if (this.type != null && event.getGenericType() == this.type.getRegistrySuperType())
```

Comment:
As a reference to `Class<T> superType` is now stored in `DeferredRegister` would it make more sense to just compare `event.getGenericType()` to that rather than querying the registry for it?

Reply:
No, because the intent is to not have superType be in here/definitive.
As one of the intentions is to be able to be used as a soft link.
By referring to the registry by name or some other method.